File: source/scene/SceneNameHolders.java
package scene;

import named.NameHolder;
import utility.abs.Allocated;

import java.io.Serializable;

public class SceneNameHolders implements Allocated, Serializable {
    public final NameHolder entity = new NameHolder();
    public final NameHolder mesh = new NameHolder();
    public final NameHolder material = new NameHolder();
    public final NameHolder texture = new NameHolder();

    public SceneNameHolders() {}

    @Override
    public void free() {
        entity.clear();
        mesh.clear();
        material.clear();
        texture.clear();
    }
}

File: source/scene/SceneNameHolders.kt
package scene

import named.NameHolder
import java.io.Serializable

class SceneNameHolders : Serializable {
    val entity: NameHolder = NameHolder()
    val mesh: NameHolder = NameHolder()
    val material: NameHolder = NameHolder()
    val texture: NameHolder = NameHolder()
}
